Doc/library/annotationlib.rst

```diff
@@ -0,0 +1,183 @@
+:mod:`!annotationlib` --- Functionality for introspecting annotations
+=====================================================================
+
+.. module:: annotationlib
+   :synopsis: Functionality for introspecting annotations
+
+
+**Source code:** :source:`Lib/annotationlib.py`
+
+.. testsetup:: default
+
+   import annotationlib
+   from annotationlib import *
+
+--------------
+
+The :mod:`!annotationlib` module provides tools for introspecting :term:`annotations <annotation>`
+on modules, classes, and functions.
+
+.. function:: call_annotate_function(annotate, format, *, owner=None)
+
+   Call the :term:`annotate function` *annotate* with the given *format*, a member of the :class:`Format`
+   enum, and return the annotations dictionary produced by the function.
+
+   Annotate functions generated by the compiler for functions, classes, and modules support only
+   the :attr:`~Format.VALUE` format when called directly. This function calls the annotate function
+   in a special environment that allows it to produce annotations in the other formats.
+
+   *owner* is the object that owns the annotation function, usually a function, class, or module.
+   If provided, it is used in the :attr:`~Format.FORWARDREF` format to produce a :class:`ForwardRef`
+   object that carry more information.
+
+   .. versionadded:: 3.14
+
+.. class:: Format
+
+   An :class:`~enum.IntEnum` describing the formats in which annotations can be returned.
+   Members of the enum, or their equivalent integer values, can be passed to
+   :func:`get_annotations` and other functions in this module, as well as to
+   :attr:`~object.__annotate__` functions.
+
+   .. attribute:: VALUE
+      :value: 1
+
+      Values are the result of evaluating the annotation expressions.
+
+   .. attribute:: FORWARDREF
+      :value: 2
+
+      Values are real annotation values (as per :attr:`Format.VALUE` format) for defined values,
+      and :class:`ForwardRef` proxies for undefined values. Real objects may be exposed to,
+      or contain references to, :class:`ForwardRef` proxy objects.
```

Comment:
What does an object being "exposed" to a ForwardRef mean here?

Reply:
I'm not sure, I got this wording from PEP 649. I'll just remove the "exposed to" part.